From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kabi breakage of mmc-block in 4.1.36
Patch-mainline: Never, openSUSE-42.1 only
References: stable-4.1.36

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/mmc/card/queue.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/mmc/card/queue.h
+++ b/drivers/mmc/card/queue.h
@@ -24,7 +24,11 @@ enum mmc_packed_type {
 
 struct mmc_packed {
 	struct list_head	list;
+#ifdef __GENKSYMS__
+	u32			cmd_hdr[1024];
+#else
 	__le32			cmd_hdr[1024];
+#endif
 	unsigned int		blocks;
 	u8			nr_entries;
 	u8			retries;
