From: jbeulich@suse.com
Subject: x86: suppress lazy MMU updates during vmalloc fault processing
Patch-mainline: Never, SUSE-Xen specific
References: bsc#951155

--- a/arch/x86/mm/fault-xen.c
+++ b/arch/x86/mm/fault-xen.c
@@ -1073,7 +1073,10 @@ __do_page_fault(struct pt_regs *regs, un
 		}
 
 		if (!(error_code & (PF_RSVD | PF_USER | PF_PROT))) {
-			if (vmalloc_fault(address) >= 0)
+			pagefault_disable(); /* suppress lazy MMU updates */
+			fault = vmalloc_fault(address);
+			pagefault_enable();
+			if (fault >= 0)
 				return;
 
 			if (kmemcheck_fault(regs, address, error_code))
